spark/src/main/scala/ai/chronon/spark/TableUtils.scala

```diff
@@ -41,6 +41,76 @@ import scala.collection.immutable
 import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
 import scala.util.{Failure, Success, Try}
 
+/**
+  * Trait to track the table format in use by a Chronon dataset and some utility methods to help
+  * retrieve metadata / configure it appropriately at creation time
+  */
+sealed trait Format {
+  // Return a sequence for partitions where each partition entry consists of a Map of partition keys to values
+  def partitions(tableName: String)(implicit sparkSession: SparkSession): Seq[Map[String, String]]
+
+  // Help specify the appropriate table type to use in the Spark create table DDL query
+  def createTableTypeString: String
+
+  // Help specify the appropriate file format to use in the Spark create table DDL query
+  def fileFormatString(format: String): String
+}
+
+case object Hive extends Format {
+  def parseHivePartition(pstring: String): Map[String, String] = {
+    pstring
+      .split("/")
+      .map { part =>
+        val p = part.split("=", 2)
+        p(0) -> p(1)
+      }
+      .toMap
+  }
+
+  override def partitions(tableName: String)(implicit sparkSession: SparkSession): Seq[Map[String, String]] = {
+    // data is structured as a Df with single composite partition key column. Every row is a partition with the
+    // column values filled out as a formatted key=value pair
+    // Eg. df schema = (partitions: String)
+    // rows = [ "day=2020-10-10/hour=00", ... ]
+    sparkSession.sqlContext
+      .sql(s"SHOW PARTITIONS $tableName")
+      .collect()
+      .map(row => parseHivePartition(row.getString(0)))
```

Comment:
Curious if the partition keys here are enforced by Chronon? i.e I'm wondering if users can specified partition key and we have increasing cardinality of this and we end up returning large entries of this

Reply:
the partition keys chosen / used here are based on the `partitionColumn` that you configure in the table util setup for your job(s). You are right that you could have a very large number of partitions here loaded up in mem.